operator/v1/types_insights.go

```diff
@@ -0,0 +1,89 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsStatus struct {
+	OperatorStatus `json:",inline"`
+	// gatherStatus provides basic information about the last Insights data gathering.
+	// When omitted, this means no data gathering has taken place yet.
+	GatherStatus *GatherStatus `json:"gatherStatus,omitempty"`
+	// reportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"reportStatus"`
+}
+
+type GatherStatus struct {
+	// lastGatherTime is the last time when Insights data gathering finished.
+	LastGatherTime metav1.Time `json:"lastGatherTime,omitempty"`
+	// startGatherTime is the time when data gathering started. The value is 0
+	// when there is no data gathering in progress.
+	StartGatherTime metav1.Time `json:"startGatherTime,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
```

Comment:
I think this should be considered as a map type with the key as the name, xref https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy
```suggestion
	// +listType=map
	// +listMapKey=name
	Gatherers []GathererStatus `json:"gatherers,omitempty"`
```

Reply:
or `atomic`? I think it can be completely replaced between gatherings. The `listType` should be specfied for every "list" right (so for `[]HealthCheck`as well)

Reply:
You could make it atomic assuming only a single owner will update the list at any one time and will calculate the entire list every time. As this is status, atomic is probably ok, but map would allow multiple controllers to observe and update the status simultaneously, depends on your backend architecture I guess

Yes, would be good to have one for `[]HealthCheck` too

Reply:
OK thanks. I think `atomic` makes slightly more sense here.